opm/autodiff/AquiferCarterTracy.hpp

```diff
@@ -0,0 +1,418 @@
+            {
+                Scalar Tc = mu_w_*phi_aq_*C_t_*r_o_*r_o_/(k_a_*c1_);
+                return Tc;
+            }
+
+            inline const Scalar aquifer_influx_constant() const
+            {
+                Scalar beta = c2_*h_*theta_*phi_aq_*C_t_*r_o_*r_o_;
+                return beta;
+            }
+
+            // This function is used to initialize and calculate the alpha_i for each grid connection to the aquifer
+            inline void initialize_connections(const Aquancon::AquanconOutput& connection)
+            {
+                const auto& eclState = ebos_simulator_.vanguard().eclState();
+                const auto& ugrid = ebos_simulator_.vanguard().grid();
+                const auto& grid = eclState.getInputGrid();
+
+                cell_idx_ = connection.global_index;
+                auto globalCellIdx = ugrid.globalCell();
+
+                assert( cell_idx_ == connection.global_index);
+                assert( (cell_idx_.size() == connection.influx_coeff.size()) );
+                assert( (connection.influx_coeff.size() == connection.influx_multiplier.size()) );
+                assert( (connection.influx_multiplier.size() == connection.reservoir_face_dir.size()) );
+
+                // We hack the cell depth values for now. We can actually get it from elementcontext pos
+                cell_depth_.resize(cell_idx_.size(), d0_);
+                alphai_.resize(cell_idx_.size(), 1.0);
+                faceArea_connected_.resize(cell_idx_.size(),0.0);
+                Scalar faceArea;
+
+                auto cell2Faces = Opm::UgGridHelpers::cell2Faces(ugrid);
+                auto faceCells  = Opm::AutoDiffGrid::faceCells(ugrid);
+
+                for (auto influxCoeff: connection.influx_coeff){
+                    std::cout << "influx_coeff = " << influxCoeff << std::endl;
+                }
+
+                for (auto influxMult: connection.influx_multiplier){
+                    std::cout << "influx_multiplier = " << influxMult << std::endl;
+                }
+
+                // Translate the C face tag into the enum used by opm-parser's TransMult class
+                Opm::FaceDir::DirEnum faceDirection;
+
+                // denom_face_areas is the sum of the areas connected to an aquifer
+                Scalar denom_face_areas = 0.;
+                for (size_t idx = 0; idx < cell_idx_.size(); ++idx)
+                {
+                    auto cellFacesRange = cell2Faces[cell_idx_.at(idx)];
+                    
+                    for(auto cellFaceIter = cellFacesRange.begin(); cellFaceIter != cellFacesRange.end(); ++cellFaceIter)
+                    {
+                        // The index of the face in the compressed grid
+                        const int faceIdx = *cellFaceIter;
+
+                        // the logically-Cartesian direction of the face
+                        const int faceTag = Opm::UgGridHelpers::faceTag(ugrid, cellFaceIter);
+
+                        
```

Comment:
Please use switch and also provide a `default` option (probably throw a run-time error. ) In the current form, `faceDirection` has the risk of not being initialized.